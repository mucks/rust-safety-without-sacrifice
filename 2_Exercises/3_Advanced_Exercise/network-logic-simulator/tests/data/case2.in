# Multi-criteria routing
NODE A weight=1 role=Observer
NODE B weight=1 role=Observer
NODE C weight=1 role=Observer
NODE D weight=1 role=Observer
LINK A B latency=5 capacity=50
LINK B C latency=5 capacity=50
LINK A C latency=15 capacity=200
LINK C D latency=7 capacity=25
LINK B D latency=50 capacity=300
ROUTE A D objective=latency
ROUTE A D objective=max_bandwidth
ROUTE A D objective=min_hops
