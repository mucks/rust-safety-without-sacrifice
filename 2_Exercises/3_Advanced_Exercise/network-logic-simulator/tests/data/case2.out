ROUTE (latency) A->D: A->B->C->D latency=17 hops=3 min_capacity=25
ROUTE (max_bandwidth) A->D: A->C->D latency=22 hops=2 min_capacity=25
ROUTE (min_hops) A->D: A->C->D latency=22 hops=2 min_capacity=25
